SpotifyAPI/SpotifyAPI.ino

```diff
@@ -0,0 +1,233 @@
+#ifndef SPOTIFY_DISPLAY
+#define SPOTIFY_DISPLAY
+
+//STD Libraries
+#include <string>
+
+// Arduino Libraries
+#include <SPI.h>
+#include <WiFiNINA.h>
+#include "base64.hpp"
+#include <ArduinoJson.h>
+
+using namespace std;
+
+class SpotifyAPIFunctionality {
+
+    public:
+        SpotifyAPIFunctionality(String clientId, String clientSecret, String refreshToken) : 
+            clientId(clientId), clientSecret(clientSecret), refreshToken(refreshToken), builtClientId(clientId + ":" + clientSecret),
+            postData("grant_type=refresh_token&refresh_token=" + refreshToken){
+                encode_base64((unsigned char *) builtClientData.c_str(), strlen(builtClientData.c_str()), base64);
+            };
+
+        // Spotify Credientials
+        String clientId;
+        String builtClientId;
+        String clientSecret;
+        String refreshToken;
+        String builtClientData;
+        unsigned char base64[128] = {0};
+        String postData;
+        String spotifyAccounts = "accounts.spotify.com";
+        String spotifyAPI = "api.spotify.com";
+
+};
+
+class SpotifyWiFiClient : public SpotifyAPIFunctionality {
```

Comment:
I think so because then SpotifyWiFiClient gets all the member variables without a ton of getters. we can make them protected but then the getters will have to be made in SpotifyWiFiClient otherwise 'loop' can't access them without getters.